raiden/transfer/identifiers.py

```diff
@@ -39,3 +32,37 @@ def validate(self) -> None:
 
         if self.channel_identifier < 0 or self.channel_identifier > constants.UINT256_MAX:
             raise ValueError("channel id is invalid")
+
+    def __str__(self) -> str:
+        return (
+            f"{self.chain_identifier}|{to_checksum_address(self.token_network_address)}|"
+            f"{self.channel_identifier}"
+        )
+
+    @staticmethod
+    def from_string(string: str) -> "CanonicalIdentifier":
+        try:
+            chain_id_str, token_network_address_hex, channel_id_str = string.split("|")
+            return CanonicalIdentifier(
+                chain_identifier=ChainID(int(chain_id_str)),
+                token_network_address=to_bytes(hexstr=token_network_address_hex),
+                channel_identifier=ChannelID(int(channel_id_str)),
+            )
+        except ValueError:
+            raise ValueError(f"Could not reconstruct canonical identifier from string: {string}")
+
+
+@dataclass(frozen=True)
+class QueueIdentifier:
+    recipient: Address
+    canonical_identifier: CanonicalIdentifier
+
+
+# According to the smart contracts as of 07/08:
+# https://github.com/raiden-network/raiden-contracts/blob/fff8646ebcf2c812f40891c2825e12ed03cc7628/raiden_contracts/contracts/TokenNetwork.sol#L213
+# channel_identifier can never be 0. We make this a requirement in the client and use this fact
+# to signify that a channel_identifier of `0` passed to the messages adds them to the
+# global queue
+CANONICAL_IDENTIFIER_GLOBAL_QUEUE = CanonicalIdentifier(
+    ChainID(0), TokenNetworkAddress(b"\1" * 20), ChannelID(0)
```

Comment:
Is this `b"\1" * 2` a dummy value? if so, do we really need this vs setting it to something valid / not setting it at all? Also, iff this is required, let's move it into a constant.

Reply:
It is a dummy value, we just need it to keep mypy happy. Mypy does not accept `TokenNetworkAddress(0)` or `None` here.